instrumentation/spymemcached-2.12/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spymemcached/CompletionListener.java

```diff
@@ -26,10 +26,12 @@
   private static final String MISS = "miss";
 
   private final Context context;
+  private final SpymemcachedRequest request;
 
   protected CompletionListener(
       Context parentContext, MemcachedConnection connection, String methodName) {
-    context = tracer().startSpan(parentContext, connection, methodName);
+    request = SpymemcachedRequest.create(connection, methodName);
```

Comment:
Can we refactor this instrumentation to insert `shouldStart()` somewhere? E.g. not register the listener at all if `shouldStart()` returns false

Reply:
We still have to create `SpymemcachedRequest` for `shouldStart`. Plus we have `CallDepth` check before creating listener. I don't think we are going to have any performance gain from this 🤔